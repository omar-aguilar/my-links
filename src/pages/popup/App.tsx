import MainContext from '@/pages/_shared/MainContext';
import RoutesContainer from './components/RoutesContainer';

const App = () => {
  return (
    <MainContext>
      <RoutesContainer />
    </MainContext>
  );
};

export default App;
