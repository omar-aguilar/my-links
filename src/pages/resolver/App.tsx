import MainContext from '@/pages/_shared/MainContext';
import Resolver from './components/Resolver';

const App = () => {
  return (
    <MainContext>
      <Resolver />
    </MainContext>
  );
};

export default App;
