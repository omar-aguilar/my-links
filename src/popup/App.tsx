import RoutesContainer from './components/RoutesContainer';

const App = () => {
  return <RoutesContainer />;
};

export default App;
