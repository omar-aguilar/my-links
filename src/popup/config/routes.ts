import type { Location } from 'history';

export type LocationState = {
  backgroundLocation?: Location;
};

type Routes = 'HOME' | 'LINK_NOT_FOUND' | 'UPDATE_LINK' | 'ADD_LINK' | 'DOMAIN_MANAGER';

type RouteConfig = {
  path: string;
  title: string;
  hideInMenu?: boolean;
};

export type RoutesConfig = Record<Routes, RouteConfig>;

export const routes: RoutesConfig = {
  HOME: {
    path: '/',
    title: 'Home',
  },
  LINK_NOT_FOUND: {
    path: '/link-not-found',
    title: 'Link Not Found',
    hideInMenu: true,
  },
  UPDATE_LINK: {
    path: '/update-link',
    title: 'Update Link',
    hideInMenu: true,
  },
  ADD_LINK: {
    path: '/add-link',
    title: 'Add Link',
  },
  DOMAIN_MANAGER: {
    path: '/manage-domains',
    title: 'Manage Domains',
  },
};

export const getRouteConfigByPath = (path: string): RouteConfig | undefined => {
  const route = Object.values(routes).find((value) => value.path === path);
  return route;
};
