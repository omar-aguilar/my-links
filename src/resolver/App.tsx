import MainContext from '../pages/common/MainContext/MainContext';
import Resolver from './components/Resolver';

const App = () => {
  return (
    <MainContext>
      <Resolver />
    </MainContext>
  );
};

export default App;
