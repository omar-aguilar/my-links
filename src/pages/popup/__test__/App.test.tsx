import { render, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { routes } from '@/pages/popup/config/routes';
import App from '@/pages/popup/App';

describe('App', () => {
  it('renders home page', async () => {
    let container: HTMLElement = document.createElement('div');
    await act(async () => {
      const component = render(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      );
      container = component.container;
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders link not found page', () => {
    const { container } = render(
      <MemoryRouter initialEntries={[routes.LINK_NOT_FOUND.path]}>
        <App />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders not found page', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/a-link-to-nowhere']}>
        <App />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
