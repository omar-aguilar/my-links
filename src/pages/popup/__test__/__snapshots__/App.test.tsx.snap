// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders home page 1`] = `
<nav
  class="mb-2 py-3 w-full border"
>
  <a
    aria-current="page"
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 bg-gray-300 rounded"
    href="/"
  >
    Home
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/add-link"
  >
    Add Link
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/manage-domains"
  >
    Manage Domains
  </a>
</nav>
`;

exports[`App renders link not found page 1`] = `
<nav
  class="mb-2 py-3 w-full border"
>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/"
  >
    Home
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/add-link"
  >
    Add Link
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/manage-domains"
  >
    Manage Domains
  </a>
</nav>
`;

exports[`App renders not found page 1`] = `
<nav
  class="mb-2 py-3 w-full border"
>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/"
  >
    Home
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/add-link"
  >
    Add Link
  </a>
  <a
    class="text-gray-800 hover:text-gray-500 text-base mr-2 py-2 px-4 hover:bg-gray-200 "
    href="/manage-domains"
  >
    Manage Domains
  </a>
</nav>
`;
